cdc/scheduler.go

```diff
@@ -278,11 +287,19 @@ func realRunProcessorWatcher(
 	info model.ChangeFeedInfo,
 	errCh chan error,
 	cb processorCallback,
-) *ProcessorWatcher {
-	sw := NewProcessorWatcher(changefeedID, captureID, pdEndpoints, etcdCli, info)
+) (*ProcessorWatcher, error) {
+	status, err := kv.GetChangeFeedStatus(ctx, etcdCli, changefeedID)
+	if err != nil && errors.Cause(err) != model.ErrChangeFeedNotExists {
+		return nil, errors.Trace(err)
+	}
+	checkpointTs := info.GetStartTs()
+	if status != nil {
+		checkpointTs = status.CheckpointTs
+	}
```

Comment:
extract a simple function for this logic, can be reused in cdc/owner.go#L551-554